### ISSUE TITLE
#[archive(repr())] can only handle one argument

### ISSUE THREAD

**Issue by Diomendius** [rkyv/rkyv#184]
```rust
use rkyv::Archive;

#[repr(C, packed)]
struct CPacked { }

#[repr(C, align(4))]
struct CAlign4 { }

#[derive(Archive)]
#[archive(repr(C))]
struct CArchivable { }

#[derive(Archive)]
#[archive(repr(C, packed))]
struct CPackedArchivable { }

#[derive(Archive)]
#[archive(repr(C, align(4)))]
struct CAlign4Archivable { }
```

Here, `CPackedArchivable` and `CAlign4Archivable` cause compiler errors because the `#[archive()]` macro requires `repr()` to have exactly one argument, unlike the native `#[repr()]`.

**Comment by djkoloski** [comment]
Thanks for the bug report, I'll get this fixed as soon as possible.

**Comment by djkoloski** [comment]
As a workaround, you should be able to use the following:

```rust
use rkyv::Archive;

#[derive(Archive)]
#[archive(repr(C))]
#[archive_attr(repr(packed))]
struct CPackedArchivable { }

#[derive(Archive)]
#[archive(repr(C))]
#[archive_attr(repr(align(4)))]
struct CAlign4Archivable { }
```

I will fix the derive macro so that you can specify the additional `repr` argument in the `archive(...)` attribute and add a warning for placing the `repr` in `archive_attr`.

**Comment by djkoloski** [comment]
Fixed by 5c2522ae0835b73eaa9d496cbf384e1d6f2b8619. This will be published later today in 0.7.7.
